File: ShoeStore/ShoeStore.Domain/Entities/RolePermission.cs
namespace ShoeStore.Domain.Entities;

public class RolePermission
{
    public byte RoleId { get; set; }
    public int PermissionId { get; set; }

    public Role Role { get; set; } = null!;
    public Permission Permission { get; set; } = null!;
}

File: shoestorefe/lib/domain/entities/role_permission.dart
class Role {
  final int id;
  final String name;
  final String code;

  Role({required this.id, required this.name, required this.code});
}

class Permission {
  final int id;
  final String code;
  final String description;

  Permission({required this.id, required this.code, required this.description});
}

class RolePermission {
  final int roleId;
  final String roleName;
  final String roleCode;
  final List<Permission> permissions;

  RolePermission({
    required this.roleId,
    required this.roleName,
    required this.roleCode,
    required this.permissions,
  });
}
